### DISCUSSION TITLE
Handling Graphql errors with Apollo

### DISCUSSION THREAD

**Discussion by DongusJr** [enisdenjo/graphql-sse#72] [Q&A]
Im setting up subscriptions with apollo using SSELink.
We have an express server. 
I followed the instructions for the server in the [documentation](https://the-guild.dev/graphql/sse/get-started#with-express) and it works as intended.

For the client side I setup an ApolloLink like instructed in the [Recipes](https://the-guild.dev/graphql/sse/recipes#with-apollo)
```ts
import {
  ApolloLink,
  Operation,
  FetchResult,
  Observable,
} from '@apollo/client/core';
import { print, GraphQLError } from 'graphql';
import { createClient, ClientOptions, Client } from 'graphql-sse';
 
class SSELink extends ApolloLink {
  private client: Client;
 
  constructor(options: ClientOptions) {
    super();
    this.client = createClient(options);
  }
 
  public request(operation: Operation): Observable<FetchResult> {
    return new Observable((sink) => {
      return this.client.subscribe<FetchResult>(
        { ...operation, query: print(operation.query) },
        {
          next: sink.next.bind(sink),
          complete: sink.complete.bind(sink),
          error: sink.error.bind(sink),
        },
      );
    });
  }
}
 
export const link = new SSELink({
  url: 'http://where.is:4000/graphql/stream',
  headers: () => {
    const session = getSession();
    if (!session) return {};
    return {
      Authorization: `Bearer ${session.token}`,
    };
  },
});
```

I defined the link this way, but additionally added `singleConnection: true` and `onMessage: console.log`.

To subscribe, I'm using an apollo useSubscription hook
```ts
const TEST_SUBSCRIPTION = gql`
  subscription TestSubscription {
    testSubscription {
      testField
    }
  }
`;

export const useTestSubscription = () => {
  const { client } = useContext(ApplicationContext);
  return useSubscription<TestSubscription>(TEST_SUBSCRIPTION, {
    client,
    onSubscriptionData: (options) => {
      console.log('options:', options);
    },
  });
};
```

When I test the "TestSubscription" event with valid data, everything is fine, I get the data passed down to the hook.
<img width="662" alt="image" src="https://github.com/enisdenjo/graphql-sse/assets/54511832/fab7c8cd-1aa9-43f4-8de0-23e45cb465ec">

However, when I try to propegate an error in the server, in my case returning `testField = undefined` when it can only be of a string type, the response is not sent down to the useSubscription hook.
<img width="662" alt="image" src="https://github.com/enisdenjo/graphql-sse/assets/54511832/34fa761c-a202-48dd-bf3e-fdc20f2b8f23">

Preferably, we would like to get the error down to the onSubscriptionData options object so we can properly handle an error when it occurs.

Is there some way for me the configure the link so that is passes the error down?

**Comment by enisdenjo** [comment]
This looks like an Apollo Client specific question. Sorry, but I don't have any experience with Apollo so I cannot help you out here.
